_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -69,121 +71,140 @@ In order to use the `s3` sink, configure AWS Identity and Access Management (IAM
-
-In cases where your data is uniform, you may be able to automatically generate a schema.
-Automatically generated schemas are based on the first event received by the codec.
-The schema will only contain keys from this event.
-Therefore, you must have all keys present in all events in order for the automatically generated schema to produce a working schema.
-Automatically generated schemas make all fields nullable.
-Use the sink's `include_keys` and `exclude_keys` configurations to control what data is included in the auto-generated schema.
+Avro fields should use a null [union](https://avro.apache.org/docs/current/specification/#unions), as this will allow missing values. Otherwise, all required fields must be present for each event. Use non-nullable fields only when you are certain they exist.
 
+Use the following options to configure the codec.
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 `schema` | Yes | String | The Avro [schema declaration](https://avro.apache.org/docs/current/specification/#schema-declaration). Not required if `auto_schema` is set to true.
 `auto_schema` | No | Boolean | When set to `true`, automatically generates the Avro [schema declaration](https://avro.apache.org/docs/current/specification/#schema-declaration) from the first event.
-
  
-### ndjson codec
-
-The `ndjson` codec writes each line as a JSON object.
+### `ndjson` codec
 
-The `ndjson` codec does not take any configurations.
+The `ndjson` codec writes each line as a JSON object. The `ndjson` codec does not take any configurations.
 
+### `json` codec
 
-### json codec
-
-The `json` codec writes events in a single large JSON file.
-Each event is written into an object within a JSON array.
+The `json` codec writes events in a single large JSON file. Each event is written into an object within a JSON array.
 
+Use the following options to configure the codec.
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 `key_name` | No | String | The name of the key for the JSON array. By default this is `events`.
 
+### `parquet` codec
 
-### parquet codec
-
-The `parquet` codec writes events into a Parquet file.
-When using the Parquet codec, set the `buffer_type` to `in_memory`.
+The `parquet` codec writes events into a Parquet file. When using the codec, set `buffer_type` to `in_memory`.
 
-The Parquet codec writes data using the Avro schema.
-Because Parquet requires an Avro schema, you may either define the schema yourself, or Data Prepper will automatically generate a schema.
-However, we generally recommend that you define your own schema so that it can best meet your needs.
+The `parquet` codec writes data using the schema. Because Parquet requires an Avro schema, you may either define the schema yourself or have Data Prepper automatically generate it. It is recommended that you define your schema to accurately reflect your needs.
```

Comment:
```suggestion
The `parquet` codec writes data using the schema. Because Parquet requires an Avro schema, you may either define the schema yourself or have Data Prepper automatically generate it. Defining your own schema is recommended, as this will allow it to be tailored to your particular use case.
```